sympy/tensor/tensor.py

```diff
@@ -2538,9 +2538,14 @@ def __iter__(self):
     def _eval_rewrite_as_Indexed(self, *args):
         return Add.fromiter(args)
 
-    def _eval_derivative(self, s):
+    def _eval_partial_derivative(self, s):
         # Evaluation like Add
-        return self.func(*[a.diff(s) for a in self.args])
+        # TODO: what about if s is a tensor and a no tensor expression?
```

Comment:
This should be handled in `TensPartialDerivative`, I wouldn't care about handling types in a hidden method.